operator/pkg/manifest/shared.go

```diff
@@ -430,8 +430,12 @@ func getClusterSpecificValues(client kube.Client, force bool, l clog.Logger) (st
 }
 
 func getFSGroupOverlay(config kube.Client) string {
-	if kube.IsAtLeastVersion(config, 19) {
-		return "values.pilot.env.ENABLE_LEGACY_FSGROUP_INJECTION=false"
+	// Set ENABLE_LEGACY_FSGROUP_INJECTION to true only for Kubernetes 1.18 or older,
+	// together with third-party-jwt, as we need the fsGroup configuration for the projected
+	// service account volume mount, which is only used by third-party-jwt.
+	jwtPolicy, _ := util.DetectSupportedJWTPolicy(config)
```

Comment:
Ah, this is actually a bit tricky.. this is only looking at the *detected* value - but users could override it. I think we may want to move `getFSGroupOverlay` to be after `getJwtTypeOverlay`, and use the result of `getJwtTypeOverlay` here?

Reply:
Done